src/go/k8s/pkg/resources/license.go

```diff
@@ -0,0 +1,114 @@
+package resources
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	redpandav1alpha1 "github.com/redpanda-data/redpanda/src/go/k8s/apis/redpanda/v1alpha1"
+	"github.com/redpanda-data/redpanda/src/go/k8s/pkg/admin"
+)
+
+type license struct {
+	client.Client
+	scheme         *runtime.Scheme
+	cluster        *redpandav1alpha1.Cluster
+	adminAPIClient admin.AdminAPIClient
+	hash           hash.Hash
+	logger         logr.Logger
+}
+
+func NewLicense(cl client.Client, scheme *runtime.Scheme, cluster *redpandav1alpha1.Cluster, aa admin.AdminAPIClient, logger logr.Logger) *license {
+	return &license{
+		Client:         cl,
+		scheme:         scheme,
+		cluster:        cluster,
+		adminAPIClient: aa,
+		hash:           sha256.New(),
+		logger:         logger,
+	}
+}
+
+// Ensure will manages license and makes sure it is loaded in the Cluster
+func (l *license) Ensure(ctx context.Context) error {
+	if l.cluster.Spec.LicenseRef == nil || l.adminAPIClient == nil {
+		l.logger.Info(
+			"Skip ensuring license, licenseRef or adminAPI internal listener not found",
+			"licenseRef", l.cluster.Spec.LicenseRef,
+			"adminAPIClient", l.adminAPIClient,
+		)
+		return nil
+	}
+
+	cond := l.cluster.Status.GetCondition(redpandav1alpha1.ClusterConfiguredConditionType)
+	if cond == nil || cond.Status != corev1.ConditionTrue {
+		return &RequeueAfterError{RequeueAfter: time.Minute * 1, Msg: "Requeue ensuring license, Cluster not yet configured"}
+	}
```

Comment:
> This PR also adds improvement in instantiating AdminAPIClient only once in the Cluster controller and passing it to funcs.

Why this is an improvement? Maybe one line comment in the code?

Reply:
This comment refers to the adminAPI instantiation in cluster controller.